### ISSUE TITLE
zeep.exceptions.LookupError: No group 'InfrastructureRootElements' in namespace urn:hl7-org:v3. Available groups are: InfrastructureRootElements

### ISSUE THREAD

**Issue by sayan801** [mvantellingen/python-zeep#360]
1. The version of zeep - zeep 1.1.0
2. The WSDL you are using - 
[pixpdq-wsdl.zip](https://github.com/mvantellingen/python-zeep/files/797081/pixpdq-wsdl.zip)

3. An example script - 
[zeep-validate-wsdl-error.py.zip](https://github.com/mvantellingen/python-zeep/files/797097/zeep-validate-wsdl-error.py.zip)

4. Complete folder containing wsdl and script for generating error.. 
[zeep-bug-report-2.zip](https://github.com/mvantellingen/python-zeep/files/797103/zeep-bug-report-2.zip)

I am using wsdl file locally.. i am not able to validate the wsdl file by running - ```python -mzeep PIXManager.wsdl``` ..  I am getting following error.. The error text is ambiguous.. It says No group InfrastructureRootElements.. But lists down InfrastructureRootElements in available groups..

````
 
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/local/lib/python2.7/dist-packages/zeep/__main__.py", line 86, in <module>
    main(args)
  File "/usr/local/lib/python2.7/dist-packages/zeep/__main__.py", line 75, in main
    client = Client(args.wsdl_file, transport=transport)
  File "/usr/local/lib/python2.7/dist-packages/zeep/client.py", line 120, in __init__
    self.wsdl = Document(wsdl, self.transport)
  File "/usr/local/lib/python2.7/dist-packages/zeep/wsdl/wsdl.py", line 65, in __init__
    root_definitions = Definition(self, document, self.location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/wsdl/wsdl.py", line 156, in __init__
    self.parse_types(doc)
  File "/usr/local/lib/python2.7/dist-packages/zeep/wsdl/wsdl.py", line 272, in parse_types
    self.types.add_documents(schema_nodes, self.location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 100, in add_documents
    document.resolve()
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 370, in resolve
    _resolve_dict(self._elements)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 364, in _resolve_dict
    new = obj.resolve()
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/elements/element.py", line 220, in resolve
    self.resolve_type()
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/elements/element.py", line 217, in resolve_type
    self.type = self.type.resolve()
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/types/complex.py", line 259, in resolve
    self._extension = self._extension.resolve()
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/types/base.py", line 80, in resolve
    return retval.resolve()
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/types/complex.py", line 246, in resolve
    self._element = self._element.resolve()
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/elements/indicators.py", line 163, in resolve
    self[i] = elm.resolve()
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/elements/references.py", line 40, in resolve
    return self._schema.get_group(self._ref)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 134, in get_group
    return self._get_instance(qname, 'get_group', 'group')
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 219, in _get_instance
    raise last_exception
zeep.exceptions.LookupError: No group 'InfrastructureRootElements' in namespace urn:hl7-org:v3. Available groups are: InfrastructureRootElements

````

**Comment by sayan801** [comment]
The difference i could notice with the other complex wsdl i have used successfully with zeep( #353 ) is - xs-group element (https://www.w3schools.com/xml/el_group.asp).. This wsdl has reference to xsd which includes xs-group elements

````
<xs:group name="InfrastructureRootElements">
      <xs:sequence>
         <xs:element xmlns="urn:hl7-org:v3" name="realmCode" type="CS" minOccurs="0"
                     maxOccurs="unbounded"/>
         <xs:element xmlns="urn:hl7-org:v3" name="typeId" type="II" minOccurs="0" maxOccurs="1"/>
         <xs:element xmlns="urn:hl7-org:v3" name="templateId" type="II" minOccurs="0"
                     maxOccurs="unbounded"/>
      </xs:sequence>
   </xs:group>
````
I could generate proxy classes using Jaxws (https://jax-ws.java.net/) .. here is the generated source and script.. This means wsdl is OK..
[proxy-class-using-jaxws-ri.zip](https://github.com/mvantellingen/python-zeep/files/800970/proxy-class-using-jaxws-ri.zip)

**Comment by mvantellingen** [comment]
@sayan801 could you give the master branch a try? It should be fixed ;-)

**Comment by sayan801** [comment]
Master branch works fine with this wsdl.. But i get following error with master when running old wsdls mentioned in #353 

````
IHE-actors/wsdl$ sudo python -mzeep xds-iti18-registry-stored-query.wsdl 
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/local/lib/python2.7/dist-packages/zeep/__main__.py", line 86, in <module>
    main(args)
  File "/usr/local/lib/python2.7/dist-packages/zeep/__main__.py", line 75, in main
    client = Client(args.wsdl_file, transport=transport)
  File "/usr/local/lib/python2.7/dist-packages/zeep/client.py", line 120, in __init__
    self.wsdl = Document(wsdl, self.transport)
  File "/usr/local/lib/python2.7/dist-packages/zeep/wsdl/wsdl.py", line 65, in __init__
    root_definitions = Definition(self, document, self.location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/wsdl/wsdl.py", line 156, in __init__
    self.parse_types(doc)
  File "/usr/local/lib/python2.7/dist-packages/zeep/wsdl/wsdl.py", line 281, in parse_types
    self.types.add_documents(schema_nodes, self.location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 102, in add_documents
    document = self.create_new_document(node, location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 189, in create_new_document
    schema.load(self, node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 355, in load
    visitor.visit_schema(node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 108, in visit_schema
    self.process(child, parent=node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 50, in process
    result = visit_func(self, node, parent)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 169, in visit_import
    schema = self.schema.create_new_document(schema_node, location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 189, in create_new_document
    schema.load(self, node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 355, in load
    visitor.visit_schema(node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 108, in visit_schema
    self.process(child, parent=node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 50, in process
    result = visit_func(self, node, parent)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 169, in visit_import
    schema = self.schema.create_new_document(schema_node, location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 189, in create_new_document
    schema.load(self, node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 355, in load
    visitor.visit_schema(node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 108, in visit_schema
    self.process(child, parent=node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 50, in process
    result = visit_func(self, node, parent)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 169, in visit_import
    schema = self.schema.create_new_document(schema_node, location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 189, in create_new_document
    schema.load(self, node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 355, in load
    visitor.visit_schema(node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 108, in visit_schema
    self.process(child, parent=node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 50, in process
    result = visit_func(self, node, parent)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 849, in visit_attribute_group
    attributes = self._process_attributes(node, children)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 926, in _process_attributes
    attribute = self.process(child, node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 50, in process
    result = visit_func(self, node, parent)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 328, in visit_attribute
    result = self.process_ref_attribute(node, array_type=array_type)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 54, in process_ref_attribute
    ref = qname_attr(node, 'ref')
  File "/usr/local/lib/python2.7/dist-packages/zeep/utils.py", line 12, in qname_attr
    return as_qname(value, node.nsmap, target_namespace)
  File "/usr/local/lib/python2.7/dist-packages/zeep/utils.py", line 21, in as_qname
    raise XMLParseError("No namespace defined for %r" % prefix)
zeep.exceptions.XMLParseError: No namespace defined for 'xml'

````

**Comment by mvantellingen** [comment]
Please provide the requested information or contact me for commercial support

**Comment by sayan801** [comment]
1. The version of zeep  -     zeep - master branch ( https://github.com/mvantellingen/python-zeep/archive/master.zip)

2. The WSDL you are using - 
[xds-iti18-registry-stored-query.wsdl.zip](https://github.com/mvantellingen/python-zeep/files/806869/xds-iti18-registry-stored-query.wsdl.zip)


3. An example script - 
[zeep-validate-wsdl-error.py.zip](https://github.com/mvantellingen/python-zeep/files/806871/zeep-validate-wsdl-error.py.zip)
t 
4. Complete folder containing wsdl and script for generating error.. 
[zeep-bug-report-360-ext.zip](https://github.com/mvantellingen/python-zeep/files/806873/zeep-bug-report-360-ext.zip)

We are using master branch of python-zeep.. When we run zeep validation function we get following error

````
'''
IHE-actors/zeep-bug-report/360-ext$ sudo python zeep-validate-wsdl-error.py 
Traceback (most recent call last):
  File "zeep-validate-wsdl-error.py", line 5, in <module>
    client = zeep.Client('xds-iti18-registry-stored-query.wsdl')
  File "/usr/local/lib/python2.7/dist-packages/zeep/client.py", line 120, in __init__
    self.wsdl = Document(wsdl, self.transport)
  File "/usr/local/lib/python2.7/dist-packages/zeep/wsdl/wsdl.py", line 65, in __init__
    root_definitions = Definition(self, document, self.location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/wsdl/wsdl.py", line 156, in __init__
    self.parse_types(doc)
  File "/usr/local/lib/python2.7/dist-packages/zeep/wsdl/wsdl.py", line 281, in parse_types
    self.types.add_documents(schema_nodes, self.location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 102, in add_documents
    document = self.create_new_document(node, location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 189, in create_new_document
    schema.load(self, node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 355, in load
    visitor.visit_schema(node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 108, in visit_schema
    self.process(child, parent=node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 50, in process
    result = visit_func(self, node, parent)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 169, in visit_import
    schema = self.schema.create_new_document(schema_node, location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 189, in create_new_document
    schema.load(self, node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 355, in load
    visitor.visit_schema(node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 108, in visit_schema
    self.process(child, parent=node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 50, in process
    result = visit_func(self, node, parent)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 169, in visit_import
    schema = self.schema.create_new_document(schema_node, location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 189, in create_new_document
    schema.load(self, node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 355, in load
    visitor.visit_schema(node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 108, in visit_schema
    self.process(child, parent=node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 50, in process
    result = visit_func(self, node, parent)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 169, in visit_import
    schema = self.schema.create_new_document(schema_node, location)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 189, in create_new_document
    schema.load(self, node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/schema.py", line 355, in load
    visitor.visit_schema(node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 108, in visit_schema
    self.process(child, parent=node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 50, in process
    result = visit_func(self, node, parent)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 849, in visit_attribute_group
    attributes = self._process_attributes(node, children)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 926, in _process_attributes
    attribute = self.process(child, node)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 50, in process
    result = visit_func(self, node, parent)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 328, in visit_attribute
    result = self.process_ref_attribute(node, array_type=array_type)
  File "/usr/local/lib/python2.7/dist-packages/zeep/xsd/visitor.py", line 54, in process_ref_attribute
    ref = qname_attr(node, 'ref')
  File "/usr/local/lib/python2.7/dist-packages/zeep/utils.py", line 12, in qname_attr
    return as_qname(value, node.nsmap, target_namespace)
  File "/usr/local/lib/python2.7/dist-packages/zeep/utils.py", line 21, in as_qname
    raise XMLParseError("No namespace defined for %r" % prefix)
zeep.exceptions.XMLParseError: No namespace defined for 'xml'
IHE-actors/zeep-bug-report/360-ext$
'''

````

This works fine with released version of zeep 1.1.0.. Master branch from github provides above mentioned error.. I am in a dead lock condition as half of my wsdl works with released version 1.1.0 and remaining works with Master branch from github...
